### PR TITLE
Test for write at start of the file

### PR DESCRIPTION
**Pull request by abhinav2603**
Integration test for testing gcsfuse if write made at the start of file is handled correctly

### CODE REVIEW

File: tools/integration_tests/implicitdir/implicitdir_test.go [hunk 1]
```diff
@@ -32,10 +32,13 @@ import (
...
-	binFile string
-	logFile string
-	mntDir  string
+	testDir  string
+	binFile  string
+	logFile  string
+	mntDir   string
+	tmpDir   string
+	fileName string
+	err      error
```
Comment (vadlakondaswetha) [inline tools/integration_tests/implicitdir/implicitdir_test.go:41]: fileName and err shouldn't be global variables. Make the fileName as return parameter for the function call. 
For the err, i can see it not be need to sent back to the caller. So you can make it a local variable inside the method.

**Review by vadlakondaswetha — COMMENTED** [review]
Update the cl description to mention that you added two test - write at start of file and writeAtEndOfFile

File: tools/integration_tests/implicitdir/write_test.go [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+
+// Provides integration tests for write flows with implicit_dir flag set.
+package implicitdir_test
+
+import (
+	"os"
+	"testing"
+)
+
+func TestAppendToEndOfFile(t *testing.T) {
```
Comment (vadlakondaswetha) [inline tools/integration_tests/implicitdir/write_test.go:23]: please make the test method names consistent. Either call it append or write in both the cases.

File: tools/integration_tests/implicitdir/implicitdir_test.go [hunk 3]
```diff
@@ -111,6 +114,40 @@ func clearKernelCache() error {
...
+}
+
+func logAndExit(s string) {
+	log.Printf(s)
+	os.Exit(1)
+}
+
+func createTempFile() {
+	// A temporary file is created and some lines are added
+	// to it for testing purposes
```
Comment (vadlakondaswetha) [inline tools/integration_tests/implicitdir/implicitdir_test.go:143]: nit: Comment is like a statement. Please start with capital letter and end with full stop.

Comment (abhinav2603) [reply tools/integration_tests/implicitdir/implicitdir_test.go:143]: Done

File: tools/integration_tests/implicitdir/write_test.go (hunk 2 above)
Comment (abhinav2603) [reply tools/integration_tests/implicitdir/write_test.go:23]: Done

File: tools/integration_tests/implicitdir/implicitdir_test.go (hunk 1 above)
Comment (abhinav2603) [reply tools/integration_tests/implicitdir/implicitdir_test.go:41]: Done

File: tools/integration_tests/implicitdir/write_test.go [hunk 4]
```diff
@@ -35,3 +35,19 @@ func TestWriteAtEndOfFile(t *testing.T) {
...
+	fileName := createTempFile()
+	f, err := os.OpenFile(fileName, os.O_WRONLY, 0600)
+	if err != nil {
+		t.Errorf("Open file for write at start: %v", err)
+	}
+
+	if _, err = f.WriteAt([]byte("line 4\n"), 0); err != nil {
+		t.Errorf("WriteString-Start: %v", err)
+	}
+
```
Comment (vadlakondaswetha) [inline tools/integration_tests/implicitdir/write_test.go:49]: in test cases, we should have the formatting like this
test initialisation
<new line>
test invocation
<new line>
assert statements.

f.close is part of test invocation only. so remove new line before f.close.

**Review by vadlakondaswetha — APPROVED** [review]

**Pull request merged**